### ISSUE TITLE
IFCRoad 

### ISSUE THREAD

**Issue by anmursas** [IfcOpenShell/IfcOpenShell#3464]
There's any example in any site that explains how to generate an IFCRoad via IFCOpenShell?

**Comment by Moult** [comment]
IfcRoads are not really any different to anything else in how they are created.

E.g. `model.createIfcRoad()` or `ifcopenshell.api.run("root.create_entity", model, ifc_class="IfcRoad")` both work.

Perhaps you're after how to create alignment based geometry?

**Comment by anmursas** [comment]
Yep I was looking to some code snippet to generate an example IFCRoad the same way there's a code example to generate an IFCWall

**Comment by Moult** [comment]
There aren't any API functions specific to roads but there's nothing stopping you from creating alignment geometry manually. Have you read https://blenderbim.org/docs-python/ifcopenshell-python/hello_world.html ?

**Comment by BIMvoice** [comment]
IfcRoad exists in IFC4X3, so create the file with that schema and add the entity, then place it in the spatial structure and give it its linear geometry via IfcAlignment. Minimal start:

```python
import ifcopenshell, ifcopenshell.api.root
model = ifcopenshell.file(schema="IFC4X3")
road = ifcopenshell.api.root.create_entity(model, ifc_class="IfcRoad", name="My Road")
```

The road's centreline/geometry is modeled with IfcAlignment (horizontal, vertical, cant) referenced from the road, and cross sections via IfcSectionedSolidHorizontal. Recommend closing.

**Comment by aothms** [comment]
Here's an example (C++) for a bridge https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.8.0/src/examples/IfcAlignment.cpp That's probably similar enough. Note that the road typically governs only the spatial structure and the elements such as IfcSlab or IfcCourse are used for the actual decks.
